packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue

```diff
@@ -366,6 +396,20 @@ const selectInputProps = computed(() => {
 });
 
 defineExpose({ input: computed(() => selectInput.value?.input) });
+watch(
+  [filteredOptions],
+  () => {
+    if (props.keepSelectionOrder) {
+      groupedOptions.value = groupByKey(filteredOptions.value, "group");
+    } else {
+      groupedOptions.value = transformGroupedData<SelectOption<TValue>, "group">(
+        groupByKey<SelectOption<TValue>, "group">(sortOptions(), "group"),
+        t.value("selections.selectGroup"),
+      );
```

Comment:
Please put this in multiple lines to make it easier to understand and read.
```suggestion
      groupedOptions.value = transformGroupedData<SelectOption<TValue>, "group">(
        groupByKey<SelectOption<TValue>, "group">(sortOptions(), "group"),
        t.value("selections.selectGroup"),
      );
```